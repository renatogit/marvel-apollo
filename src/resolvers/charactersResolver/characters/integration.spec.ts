const CharactersResolver = require('.');
const DataSourcesCharactersAPI = require('@/dataSources/characters');
const ConfigServer = require('@/configs/test');
const QUERY_GET_CHARACTERS = require('@query/characters/queryCharacters');
const MOCK_GET_CHARACTERS = require('@mock/characters/mockCharacters');

describe('resolvers/charactersResolver/integration', () => {
	it('fetches characters and generate snapshots', async () => {
		const {server} = await ConfigServer(
			DataSourcesCharactersAPI,
			CharactersResolver,
			MOCK_GET_CHARACTERS
		);

		const {body} = await server.executeOperation({
			query: QUERY_GET_CHARACTERS,
		});

		expect(body).toMatchSnapshot();
	});
});
