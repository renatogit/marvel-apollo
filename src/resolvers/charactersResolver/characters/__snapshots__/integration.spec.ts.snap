// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolvers/charactersResolver/integration fetches characters and generate snapshots 1`] = `
{
  "kind": "single",
  "singleResult": {
    "data": {
      "characters": {
        "code": 200,
        "copyright": "mock data",
        "data": {
          "count": 20,
          "limit": 20,
          "offset": 0,
          "results": [
            {
              "comics": {
                "available": 12,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 12,
              },
              "description": "mock data",
              "events": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "id": 1011334,
              "modified": "mock data",
              "name": "mock data",
              "resourceURI": "mock data",
              "series": {
                "name": "mock data",
                "resourceURI": "mock data",
              },
              "stories": {
                "available": 21,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 20,
              },
              "thumbnail": {
                "extension": "mock data",
                "path": "mock data",
              },
              "urls": [
                {
                  "type": "mock data",
                  "url": "mock data",
                },
                {
                  "type": "mock data",
                  "url": "mock data",
                },
              ],
            },
            {
              "comics": {
                "available": 12,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 12,
              },
              "description": "mock data",
              "events": {
                "available": 1,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 1,
              },
              "id": 1017100,
              "modified": "mock data",
              "name": "mock data",
              "resourceURI": "mock data",
              "series": {
                "name": "mock data",
                "resourceURI": "mock data",
              },
              "stories": {
                "available": 21,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 20,
              },
              "thumbnail": {
                "extension": "mock data",
                "path": "mock data",
              },
              "urls": [
                {
                  "type": "mock data",
                  "url": "mock data",
                },
                {
                  "type": "mock data",
                  "url": "mock data",
                },
              ],
            },
          ],
          "total": 1564,
        },
        "etag": "mock data",
        "status": "mock data",
      },
    },
    "errors": undefined,
  },
}
`;
